cmd/thv/app/rm.go

```diff
@@ -37,9 +38,37 @@ func rmCmdFunc(cmd *cobra.Command, args []string) error {
 	}
 
 	// Delete container.
-	if err := manager.DeleteContainer(ctx, containerName, rmForce); err != nil {
+	if err := manager.DeleteContainer(ctx, containerName, rmForce, true); err != nil {
 		return fmt.Errorf("failed to delete container: %v", err)
 	}
 
+	// Delete associated egress container.
+	egressContainerName := containerName + "-egress"
+	if err := manager.DeleteContainer(ctx, egressContainerName, rmForce, false); err != nil {
+		// just log the error and continue
+		logger.Warnf("failed to delete egress container %q: %v", egressContainerName, err)
+	}
+
+	// Delete networks if there are no containers using them.
+	toolHiveContainers, err := manager.ListContainers(ctx, listAll)
+	if err != nil {
+		return fmt.Errorf("failed to list containers: %v", err)
+	}
+
+	// Delete associated internal network
+	networkName := "toolhive-" + containerName + "-internal"
+	if err := manager.DeleteNetwork(ctx, networkName); err != nil {
+		// just log the error and continue
+		logger.Warnf("failed to delete network %q: %v", networkName, err)
+	}
+
+	if len(toolHiveContainers) == 0 {
+		// remove external network
+		if err := manager.DeleteNetwork(ctx, "toolhive-external"); err != nil {
```

Comment:
This is also an implementation detail, I think it should just go in the DeleteContainer call